tutormfe/templates/mfe/build/mfe/Dockerfile

```diff
@@ -20,60 +24,65 @@ FROM base AS i18n
+RUN stat /openedx/app/src/i18n/messages 2> /dev/null || (echo "missing messages folder" && mkdir -p /openedx/app/src/i18n/messages)
+RUN /openedx/i18n/i18n-merge.js /openedx/app/src/i18n/messages /openedx/i18n/{{ app_name }} /openedx/app/src/i18n/messages
 
-######## {{ app["name"] }} (common)
-FROM base AS {{ app["name"] }}-common
-COPY --from={{ app["name"] }}-src /openedx/app/package.json /openedx/app/package.json
-COPY --from={{ app["name"] }}-src /openedx/app/package-lock.json /openedx/app/package-lock.json
+######## {{ app_name }} (common)
+FROM base AS {{ app_name }}-common
+COPY --from={{ app_name }}-src /package.json /openedx/app/package.json
+COPY --from={{ app_name }}-src /package-lock.json /openedx/app/package-lock.json
 ARG NPM_REGISTRY={{ NPM_REGISTRY }}
 {{ patch("mfe-dockerfile-pre-npm-install") }}
 {#- Required for building optipng on M1 #}
 ENV CPPFLAGS=-DPNG_ARM_NEON_OPT=0
 {#- We define this environment variable to bypass an issue with the installation of pact https://github.com/pact-foundation/pact-js-core/issues/264 #}
 ENV PACT_SKIP_BINARY_INSTALL=true
-RUN npm clean-install --no-audit --no-fund --registry=$NPM_REGISTRY \
-  && rm -rf ~/.npm
+RUN {% if is_buildkit_enabled() %}--mount=type=cache,target=/root/.npm,sharing=shared {% endif %}npm clean-install --no-audit --no-fund --registry=$NPM_REGISTRY
 {{ patch("mfe-dockerfile-post-npm-install") }}
-COPY --from={{ app["name"] }}-src /openedx/app /openedx/app
-COPY --from={{ app["name"] }}-i18n /openedx/app/src/i18n/messages /openedx/app/src/i18n/messages
+COPY --from={{ app_name }}-src / /openedx/app
+COPY --from={{ app_name }}-i18n /openedx/app/src/i18n/messages /openedx/app/src/i18n/messages
 EXPOSE {{ app['port'] }}
 
 # Configuration needed at build time
-ENV APP_ID={{ app["name"] }}
-ENV PUBLIC_PATH='/{{ app["name"] }}/'
+ENV APP_ID={{ app_name }}
+ENV PUBLIC_PATH='/{{ app_name }}/'
 # We could in theory point the mfe_config API directly to the LMS. But for that we would
 # have to code the LMS url into the mfe image, and this configuration is user-dependent.
 # So we point to a relative url that will be a proxy for the LMS.
 ENV MFE_CONFIG_API_URL=/api/mfe_config/v1
 ARG ENABLE_NEW_RELIC=false
 
-######## {{ app["name"] }} (dev)
-FROM {{ app["name"] }}-common AS {{ app["name"] }}-dev
+######## {{ app_name }} (dev)
+FROM {{ app_name }}-common AS {{ app_name }}-dev
 ENV NODE_ENV=development
 CMD ["/bin/bash", "-c", "npm run start --- --config ./webpack.dev-tutor.config.js"]
 {% endfor %}
 
 # Production images are last to accelerate dev image building
-{%- for app in iter_values_named(suffix="MFE_APP") %}
-######## {{ app["name"] }} (production)
-FROM {{ app["name"] }}-common AS {{ app["name"] }}-prod
+{%- for app_name, app in iter_mfes() %}
+######## {{ app_name }} (production)
+FROM {{ app_name }}-common AS {{ app_name }}-prod
 ENV NODE_ENV=production
```

Comment:
Cna you we add here support the new feuture 
`ARG USE_PURGECSS=false`
`ENV USE_PURGECESS=$USE_PURGECSS`
So that it can set at build, i.e. `docker build mfe --build-arg USE_PURGECESS=true`
ref: https://github.com/openedx/frontend-build#optimization

Reply:
I just tested building the mfe image with `USE_PURGECSS=true` and now I think I know your pain when you build on macOS and Docker eats all your memory.

I'm reluctant to introduce a build-time argument for such a specific use case. Instead, can I suggest we add a `{{ patch("mfe-dockerfile-prod-pre-build") }}` statement?

Reply:
Indeed I don't meant to have it by default,  just to have it as env-variable would just it make it easier for people to use, it, _an extra cli arg vs creating a plguin file_. Secondaly regarding resources extensive #125, which is an issue even without using prugeCSS, https://discuss.openedx.org/t/npm-err-code-err-socket-timeout-when-building-mfe-image/10414. Lastly regarding the diff in file size, I would just add that it's not directly about the file size/request size, as it's about perfoamnce; as it's about the time it takes the browser to read compile the file content.

Reply:
I really prefer a simple plugin over a new build arg or tutor setting. The new patch would also have the advantage that it could also be used for other purposes.